diff -urN openser-1.2.2-tls.orig/Makefile openser-1.2.2-tls/Makefile
--- openser-1.2.2-tls.orig/Makefile	2007-08-16 16:23:20.000000000 +0200
+++ openser-1.2.2-tls/Makefile	2007-09-19 15:15:01.000000000 +0200
@@ -318,12 +318,12 @@
 	scripts/mysqldb.sh create
 	-@echo "Done"
 
-mk-install-dirs: $(cfg-prefix)/$(cfg-dir) $(bin-prefix)/$(bin-dir) \
+mk-install-dirs: $(cfg-dir) $(bin-prefix)/$(bin-dir) \
 			$(modules-prefix)/$(modules-dir) $(doc-prefix)/$(doc-dir) \
 			$(man-prefix)/$(man-dir)/man8 $(man-prefix)/$(man-dir)/man5
 
-$(cfg-prefix)/$(cfg-dir): 
-		mkdir -p $(cfg-prefix)/$(cfg-dir)
+$(cfg-dir): 
+		mkdir -p $(cfg-dir)
 
 $(bin-prefix)/$(bin-dir):
 		mkdir -p $(bin-prefix)/$(bin-dir)
@@ -342,43 +342,43 @@
 		mkdir -p $(man-prefix)/$(man-dir)/man5
 		
 # note: on solaris 8 sed: ? or \(...\)* (a.s.o) do not work
-install-cfg: $(cfg-prefix)/$(cfg-dir)
+install-cfg: $(cfg-dir)
 		sed -e "s#/usr/.*lib/$(NAME)/modules/#$(modules-target)#g" \
-			< etc/$(NAME).cfg > $(cfg-prefix)/$(cfg-dir)$(NAME).cfg.sample0
+			< etc/$(NAME).cfg > $(cfg-dir)$(NAME).cfg.sample0
 		sed -e "s#/usr/.*etc/$(NAME)/tls/#$(cfg-target)tls/#g" \
-			< $(cfg-prefix)/$(cfg-dir)$(NAME).cfg.sample0 \
-			> $(cfg-prefix)/$(cfg-dir)$(NAME).cfg.sample
-		rm -fr $(cfg-prefix)/$(cfg-dir)$(NAME).cfg.sample0
-		chmod 644 $(cfg-prefix)/$(cfg-dir)$(NAME).cfg.sample
+			< $(cfg-dir)$(NAME).cfg.sample0 \
+			> $(cfg-dir)$(NAME).cfg.sample
+		rm -fr $(cfg-dir)$(NAME).cfg.sample0
+		chmod 644 $(cfg-dir)$(NAME).cfg.sample
 		if [ -z "${skip_cfg_install}" -a \
-				! -f $(cfg-prefix)/$(cfg-dir)$(NAME).cfg ]; then \
-			mv -f $(cfg-prefix)/$(cfg-dir)$(NAME).cfg.sample \
-				$(cfg-prefix)/$(cfg-dir)$(NAME).cfg; \
+				! -f $(cfg-dir)$(NAME).cfg ]; then \
+			mv -f $(cfg-dir)$(NAME).cfg.sample \
+				$(cfg-dir)$(NAME).cfg; \
 		fi
 		# radius dictionary
-		$(INSTALL_TOUCH) $(cfg-prefix)/$(cfg-dir)/dictionary.radius
-		$(INSTALL_CFG) etc/dictionary.radius $(cfg-prefix)/$(cfg-dir)
+		$(INSTALL_TOUCH) $(cfg-dir)/dictionary.radius
+		$(INSTALL_CFG) etc/dictionary.radius $(cfg-dir)
 		# openserctl config
-		$(INSTALL_TOUCH)   $(cfg-prefix)/$(cfg-dir)/openserctlrc.sample
+		$(INSTALL_TOUCH) $(cfg-dir)/openserctlrc.sample
 		$(INSTALL_CFG) scripts/openserctlrc \
-			$(cfg-prefix)/$(cfg-dir)/openserctlrc.sample
-		if [ ! -f $(cfg-prefix)/$(cfg-dir)/openserctlrc ]; then \
-			mv -f $(cfg-prefix)/$(cfg-dir)/openserctlrc.sample \
-				$(cfg-prefix)/$(cfg-dir)/openserctlrc; \
+			$(cfg-dir)/openserctlrc.sample
+		if [ ! -f $(cfg-dir)/openserctlrc ]; then \
+			mv -f $(cfg-dir)/openserctlrc.sample \
+				$(cfg-dir)/openserctlrc; \
 		fi
-		#$(INSTALL_CFG) etc/$(NAME).cfg $(cfg-prefix)/$(cfg-dir)
+		#$(INSTALL_CFG) etc/$(NAME).cfg $(cfg-dir)
 		if [ "$(TLS)" != "" ] ; then \
-			mkdir -p $(cfg-prefix)/$(cfg-dir)/tls ; \
-			mkdir -p $(cfg-prefix)/$(cfg-dir)/tls/rootCA ; \
-			mkdir -p $(cfg-prefix)/$(cfg-dir)/tls/rootCA/certs ; \
-			mkdir -p $(cfg-prefix)/$(cfg-dir)/tls/rootCA/private ; \
-			mkdir -p $(cfg-prefix)/$(cfg-dir)/tls/user ; \
+			mkdir -p $(cfg-dir)/tls ; \
+			mkdir -p $(cfg-dir)/tls/rootCA ; \
+			mkdir -p $(cfg-dir)/tls/rootCA/certs ; \
+			mkdir -p $(cfg-dir)/tls/rootCA/private ; \
+			mkdir -p $(cfg-dir)/tls/user ; \
 			for FILE in $(tls_configs) ; do \
 				if [ -f etc/$$FILE ] ; then \
 					$(INSTALL_TOUCH) etc/$$FILE \
-						$(cfg-prefix)/$(cfg-dir)/$$FILE ; \
+						$(cfg-dir)/$$FILE ; \
 					$(INSTALL_CFG) etc/$$FILE \
-						$(cfg-prefix)/$(cfg-dir)/$$FILE ; \
+						$(cfg-dir)/$$FILE ; \
 				fi ;\
 			done ; \
 		fi
diff -urN openser-1.2.2-tls.orig/Makefile.defs openser-1.2.2-tls/Makefile.defs
--- openser-1.2.2-tls.orig/Makefile.defs	2007-08-16 16:23:20.000000000 +0200
+++ openser-1.2.2-tls/Makefile.defs	2007-09-19 15:09:37.000000000 +0200
@@ -124,7 +124,7 @@
 SYSBASE ?= /usr
 
 # dirs 
-cfg-dir = etc/$(MAIN_NAME)/
+cfg-dir = $(basedir)/etc/$(MAIN_NAME)/
 bin-dir = sbin/
 
 ARCH_B= $(shell echo $(ARCH) | sed -e 's/.*64.*/64b/')
@@ -195,7 +195,7 @@
 
 
 # target dirs for various stuff
-cfg-target = $(prefix)/$(cfg-dir)
+cfg-target = $(cfg-dir)
 bin-target = $(prefix)/$(bin-dir)
 lib-target = $(prefix)/$(lib-dir)
 modules-target = $(prefix)/$(modules-dir)
